internal/refreshtoken/refresh_token_test.go

```diff
@@ -0,0 +1,398 @@
+				LastItemUpdatedTime: fiveDaysAgo,
+			},
+			grantsHash:    []byte("some hash"),
+			resourceType:  resource.Target,
+			wantErrString: "refresh token was updated before its creation time",
+			wantErrCode:   errors.InvalidParameter,
+		},
+		{
+			name: "updated after now",
+			token: &refreshtoken.Token{
+				CreatedTime:         fiveDaysAgo,
+				UpdatedTime:         time.Now().AddDate(0, 0, 1),
+				ResourceType:        resource.Target,
+				GrantsHash:          []byte("some hash"),
+				LastItemId:          "s_1234567890",
+				LastItemUpdatedTime: fiveDaysAgo,
+			},
+			grantsHash:    []byte("some hash"),
+			resourceType:  resource.Target,
+			wantErrString: "refresh token was updated in the future",
+			wantErrCode:   errors.InvalidParameter,
+		},
+		{
+			name: "resource type mismatch",
+			token: &refreshtoken.Token{
+				CreatedTime:         fiveDaysAgo,
+				UpdatedTime:         fiveDaysAgo.AddDate(0, 0, 1),
+				ResourceType:        resource.Target,
+				GrantsHash:          []byte("some hash"),
+				LastItemId:          "s_1234567890",
+				LastItemUpdatedTime: fiveDaysAgo,
+			},
+			grantsHash:    []byte("some hash"),
+			resourceType:  resource.SessionRecording,
+			wantErrString: "refresh token resource type does not match expected resource type",
+			wantErrCode:   errors.InvalidParameter,
+		},
+		{
+			name: "last item ID unset",
+			token: &refreshtoken.Token{
+				CreatedTime:         fiveDaysAgo,
+				UpdatedTime:         fiveDaysAgo.AddDate(0, 0, 1),
+				ResourceType:        resource.Target,
+				GrantsHash:          []byte("some hash"),
+				LastItemId:          "",
+				LastItemUpdatedTime: fiveDaysAgo,
+			},
+			grantsHash:    []byte("some hash"),
+			resourceType:  resource.Target,
+			wantErrString: "refresh token missing last item ID",
+			wantErrCode:   errors.InvalidParameter,
+		},
+		{
+			name: "last item ID unset",
+			token: &refreshtoken.Token{
+				CreatedTime:         fiveDaysAgo,
+				UpdatedTime:         fiveDaysAgo.AddDate(0, 0, 1),
+				ResourceType:        resource.Target,
+				GrantsHash:          []byte("some hash"),
+				LastItemId:          "",
+				LastItemUpdatedTime: fiveDaysAgo,
+			},
+			grantsHash:    []byte("some hash"),
+			resourceType:  resource.Target,
+			wantErrString: "refresh token missing last item ID",
+			wantErrCode:   errors.InvalidParameter,
+		},
+		{
+			name: "updated in the future",
+			token: &refreshtoken.Token{
+				CreatedTime:         fiveDaysAgo,
+				UpdatedTime:         fiveDaysAgo.AddDate(0, 0, 1),
+				ResourceType:        resource.Target,
+				GrantsHash:          []byte("some hash"),
+				LastItemId:          "s_1234567890",
+				LastItemUpdatedTime: time.Now().AddDate(1, 0, 0),
+			},
+			grantsHash:    []byte("some hash"),
+			resourceType:  resource.Target,
+			wantErrString: "refresh token last item was updated in the future",
+			wantErrCode:   errors.InvalidParameter,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Yes 😰. Fortunately, in this case, it doesn't matter, since the tests are executed serially, but adding `t.Parallel` here would break it without a fix.